### PR TITLE
Mesh_Primitives refactor 

### PR DESCRIPTION
**Pull request by stevk**

### CODE REVIEW

File: Source/ModelGroups/Mesh_Primitives.cs [hunk 1]
```diff
@@ -1,193 +1,114 @@
...
-            };
-            Vector4 colors0 = new Vector4(0.0f, 1.0f, 0.0f, 1.0f);
-            Vector4 colors1 = new Vector4(0.0f, 0.0f, 1.0f, 1.0f);
+                var properties = new List<Property>();
+                var meshPrimitives = MeshPrimitive.CreateMultiPrimitivePlane();
+                foreach (var primitive in meshPrimitives)
+                {
+                    primitive.TextureCoordSets = null;
+                }
+                meshPrimitives[0].Material = new Runtime.Material()
```
Comment (bghgary) [inline Source/ModelGroups/Mesh_Primitives.cs]: This seems more complicated than it should be. Let's discuss offline.

Comment (bghgary) [reply Source/ModelGroups/Mesh_Primitives.cs]: Create the materials in the helper functions instead of here.

Comment (stevk) [reply Source/ModelGroups/Mesh_Primitives.cs]: Fixed

File: Source/ModelGroup.cs [hunk 2]
```diff
@@ -115,8 +115,19 @@ public static List<int> GetSinglePlaneIndices()
                 };
             }
 
-            public static List<Runtime.MeshPrimitive> CreateMultiPrimitivePlane()
+            public static List<Runtime.MeshPrimitive> CreateMultiPrimitivePlane(bool includeTextureCoords = true)
             {
+                List<List<List<Vector2>>> textureCoordSets = new List<List<List<Vector2>>>()
```
Comment (bghgary) [inline Source/ModelGroup.cs]: This still seems weird to me. How about split GetMultiPrimitivePlaneTextureCoordSets into two and set the TextureCoordSets only when includeTextureCoords is true?

Comment (stevk) [reply Source/ModelGroup.cs]: It is already only being set when includeTextureCoords is true. You mean to not do the set null bit?

Comment (bghgary) [reply Source/ModelGroup.cs]: I mean why create an array when you are only going to use it as separated entries.

**Pull request merged**